app/src/user/java/mainFunctionality/AccountFragment.java

```diff
@@ -74,66 +77,56 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+        setOnFocusChangeListenerForModifActions(name);
+
         AutoCompleteTextView surname = view.findViewById(R.id.userLastNameMyAccount);
+        setOnFocusChangeListenerForModifActions(surname);
+
         TextView email = view.findViewById(R.id.userEmailMyAccount);
-        Button modifyAccount = view.findViewById(R.id.btn_modify_account);
+        Button modifyPassword = view.findViewById(R.id.btn_modify_password);
         Button removeAccount = view.findViewById(R.id.btn_remove_account);
         Button confirmModifs = view.findViewById(R.id.btn_confirm_user_modification);
-        Button cancelModifs = view.findViewById(R.id.btn_cancel_user_modification);
-        LinearLayout accountActions = view.findViewById(R.id.account_actions);
-        LinearLayout accountModifs = view.findViewById(R.id.account_modifications);
-
-        try {
-            User user = usersModel.getUser();
-            name.append(user.getName());
-            surname.append(user.getSurname());
-            email.append(user.getEmail().toLowerCase());
-        } catch (BackendException | BackendConnectionException be) {
-            showErrorDialog(getActivity(), be.getMessage());
-        }
 
-        modifyAccount.setOnClickListener(new View.OnClickListener() {
+        name.append(dbUser.getName());
+        surname.append(dbUser.getSurname());
+        email.append(user.getEmail());
+
+        modifyPassword.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                accountActions.setVisibility(View.GONE);
-                accountModifs.setVisibility(View.VISIBLE);
-                name.setEnabled(true);
-                surname.setEnabled(true);
+                setFragment(new UpdatePasswordFragment(), true);
             }
         });
 
         confirmModifs.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
-                builder.setMessage("Desea modificar el usuario?")
-                        .setPositiveButton("Aceptar", new DialogInterface.OnClickListener() {
-                            @Override
-                            public void onClick(DialogInterface dialog, int position1) {
-                                try {
-                                    usersModel.modifyUser(name.getText().toString(),
-                                            surname.getText().toString(), user.getEmail());
-                                } catch (BackendException | BackendConnectionException |
-                                        FailedToModifyUserException e) {
-                                    dialog.dismiss();
-                                    showErrorDialog(getActivity(), e.getMessage());
+                if (!(isSameText(name, dbUser.getName()) && isSameText(surname, dbUser.getSurname()))) {
```

Comment:
Sólo se puede cambiar el nombre pero no el apellido?

Reply:
Creo que estás mirando mal los paréntesis, es un not(nameEsIgual Y apellidoEsIgual), o sea, si alguno de los dos es distinto.